src/cluster.c

```diff
@@ -1216,6 +1216,36 @@ unsigned int keyHashSlot(char *key, int keylen) {
     return crc16(key+s+1,e-s-1) & 0x3FFF;
 }
 
+/* If it can be inferred that the given glob-style pattern, as implemented in
+ * stringmatchlen() in util.c, only can match keys belonging to a single slot,
+ * that slot is returned. Otherwise -1 is returned. */
+int patternHashSlot(char *pattern, int length) {
+    int s = -1; /* index of the first '{' */
+
+    for (int i = 0; i < length; i++) {
+        if (pattern[i] == '*' || pattern[i] == '?' || pattern[i] == '[') {
```

Comment:
Wild card could be present after the hashtag though and server would still restrict it that slot, right?

Example:

Full scan
```
127.0.0.1:6379> SCAN 0
1) "9126805742"
2) 1) "{000010072290}a:000090094024"
   2) "{000042722283}a:000087920779"
   3) "{000045177331}a:000022628153"
   4) "{000098432023}a:000027608496"
   5) "{000009479443}a:000071709653"
   6) "{000082673023}a:000079869437"
   7) "{000006315737}a:000005009014"
   8) "{000027213473}a:000062063671"
```

Restricted scan with wildcard glob match
```
127.0.0.1:6379> SCAN 0 MATCH {000010072290}a:*
1) "9126805742"
2) 1) "{000010072290}a:000090094024"
```

Reply:
In a pattern like `"{000010072290}a:*"`, we don't return -1. We return the slot when `}` is found, before we have seen the wildcard. It is the last `else if` after this `if`.

```C
        } else if (s >= 0 && pattern[i] == '}') {
            /* Non-empty tag '{...}' found. Hash what's between braces. */
            return crc16(pattern + s + 1, i - s - 1) & 0x3FFF;
        }
```